### ISSUE TITLE
WebRTC answer does not reflect offer's media id tags

### ISSUE THREAD

**Issue by grondinjc** [sipsorcery-org/sipsorcery#609]
Hello 😀,

When using a xamarin app as a caller and sipsorcery as a callee, the session is not fully established. On the xamarin side, I get the following error `"Failed to set remote answer sdp: The order of m-lines in answer doesn't match order in offer. Rejecting answer.`

### The offer generated by xamarin : 
```
v=0
o=- 1064364449942365659 2 IN IP4 127.0.0.1
s=-
t=0 0
a=group:BUNDLE audio video
a=msid-semantic: WMS stream0
m=audio 9 UDP/TLS/RTP/SAVPF 111 103 104 9 102 0 8 106 105 13 110 112 113 126
c=IN IP4 0.0.0.0
a=rtcp:9 IN IP4 0.0.0.0
a=ice-ufrag:G5P/
a=ice-pwd:FICf2eBzvl5r/O/uf1ktSyuc
a=ice-options:trickle renomination
a=fingerprint:sha-256 5D:03:7C:22:69:2E:E7:10:17:5F:31:86:E6:47:2F:6F:1D:4C:A6:BF:5B:DE:0C:FB:8A:17:15:AA:22:63:0C:FD
a=setup:actpass
a=mid:audio
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=extmap:3 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=sendrecv
a=rtcp-mux
a=rtpmap:111 opus/48000/2
a=rtcp-fb:111 transport-cc
a=fmtp:111 minptime=10;useinbandfec=1
a=rtpmap:103 ISAC/16000
a=rtpmap:104 ISAC/32000
a=rtpmap:9 G722/8000
a=rtpmap:102 ILBC/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:106 CN/32000
a=rtpmap:105 CN/16000
a=rtpmap:13 CN/8000
a=rtpmap:110 telephone-event/48000
a=rtpmap:112 telephone-event/32000
a=rtpmap:113 telephone-event/16000
a=rtpmap:126 telephone-event/8000
a=ssrc:3780525913 cname:FLLo3gHcblO+MbrR
a=ssrc:3780525913 msid:stream0 audio0
a=ssrc:3780525913 mslabel:stream0
a=ssrc:3780525913 label:audio0
m=video 9 UDP/TLS/RTP/SAVPF 96 97 98 99 100 101 127
c=IN IP4 0.0.0.0
a=rtcp:9 IN IP4 0.0.0.0
a=ice-ufrag:G5P/
a=ice-pwd:FICf2eBzvl5r/O/uf1ktSyuc
a=ice-options:trickle renomination
a=fingerprint:sha-256 5D:03:7C:22:69:2E:E7:10:17:5F:31:86:E6:47:2F:6F:1D:4C:A6:BF:5B:DE:0C:FB:8A:17:15:AA:22:63:0C:FD
a=setup:actpass
a=mid:video
a=extmap:14 urn:ietf:params:rtp-hdrext:toffset
a=extmap:2 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=extmap:13 urn:3gpp:video-orientation
a=extmap:3 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=extmap:5 http://www.webrtc.org/experiments/rtp-hdrext/playout-delay
a=extmap:6 http://www.webrtc.org/experiments/rtp-hdrext/video-content-type
a=extmap:7 http://www.webrtc.org/experiments/rtp-hdrext/video-timing
a=extmap:8 http://www.webrtc.org/experiments/rtp-hdrext/color-space
a=sendrecv
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:96 VP8/90000
a=rtcp-fb:96 goog-remb
a=rtcp-fb:96 transport-cc
a=rtcp-fb:96 ccm fir
a=rtcp-fb:96 nack
a=rtcp-fb:96 nack pli
a=rtpmap:97 rtx/90000
a=fmtp:97 apt=96
a=rtpmap:98 VP9/90000
a=rtcp-fb:98 goog-remb
a=rtcp-fb:98 transport-cc
a=rtcp-fb:98 ccm fir
a=rtcp-fb:98 nack
a=rtcp-fb:98 nack pli
a=rtpmap:99 rtx/90000
a=fmtp:99 apt=98
a=rtpmap:100 red/90000
a=rtpmap:101 rtx/90000
a=fmtp:101 apt=100
a=rtpmap:127 ulpfec/90000
a=ssrc-group:FID 3851740345 4165955869
a=ssrc:3851740345 cname:FLLo3gHcblO+MbrR
a=ssrc:3851740345 msid:stream0 video0
a=ssrc:3851740345 mslabel:stream0
a=ssrc:3851740345 label:video0
a=ssrc:4165955869 cname:FLLo3gHcblO+MbrR
a=ssrc:4165955869 msid:stream0 video0
a=ssrc:4165955869 mslabel:stream0
a=ssrc:4165955869 label:video0
```

### The answer generated by sipsorcery
```
v=0
o=- 30500 0 IN IP4 127.0.0.1
s=sipsorcery
t=0 0
a=group:BUNDLE 0 1
m=audio 9 UDP/TLS/RTP/SAVPF 9 8 13 126
c=IN IP4 0.0.0.0
a=ice-ufrag:OJNI
a=ice-pwd:PLNHUKOIRKDDEGDLECTYEKTU
a=fingerprint:sha-256 DD:03:20:57:4F:55:A5:27:83:AC:5E:09:44:E6:01:68:83:2E:29:16:43:E2:B8:A5:42:35:38:63:FF:09:EC:F1
a=setup:active
a=candidate:2572 1 udp 659136 192.168.0.127 60304 typ host generation 0
a=ice-options:ice2,trickle
a=mid:0
a=rtpmap:9 G722/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:13 CN/8000
a=rtpmap:126 telephone-event/8000
a=rtcp-mux
a=rtcp:9 IN IP4 0.0.0.0
a=end-of-candidates
a=sendrecv
a=ssrc:1698763858 cname:0b7daea3-9d0b-401c-a634-7fb22bb8bb46
m=video 9 UDP/TLS/RTP/SAVPF 96 98
c=IN IP4 0.0.0.0
a=ice-ufrag:OJNI
a=ice-pwd:PLNHUKOIRKDDEGDLECTYEKTU
a=fingerprint:sha-256 DD:03:20:57:4F:55:A5:27:83:AC:5E:09:44:E6:01:68:83:2E:29:16:43:E2:B8:A5:42:35:38:63:FF:09:EC:F1
a=setup:active
a=ice-options:ice2,trickle
a=mid:1
a=rtpmap:96 VP8/90000
a=rtpmap:98 VP9/90000
a=rtcp-mux
a=rtcp:9 IN IP4 0.0.0.0
a=sendrecv
a=ssrc:1168494149 cname:5ded661f-8bad-45cd-b06b-03eed4957bc4
``` 

# Summary 

The main issue of the problem is that sipsorcery does not reuse the media id tag specified in the offer. Therefore the issue is that the offer is : 
```
...
a=group:BUNDLE audio video
...
a=mid:audio
...
a=mid:video
```

and the answer is :
```
...
a=group:BUNDLE 0 1
...
a=mid:0
...
a=mid:1
```

**Comment by grondinjc** [comment]
I tried a hack to replicate the media id tags, the session is established perfectly and media goes throught just fine.
The hack involved replacing strings of the answer's SDP within the RTCSessionDescriptionInit.

```
var init = connection.createAnswer(answerOptions);
init.sdp = init.sdp.Replace("a=group:BUNDLE 0 1", "a=group:BUNDLE audio video").Replace("a=mid:0", "a=mid:audio").Replace("a=mid:1", "a=mid:video");
```

**Comment by grondinjc** [comment]
Of what I've seen, we are index-based instead of string-based : 
https://github.com/sipsorcery-org/sipsorcery/blob/master/src/net/WebRTC/RTCPeerConnection.cs#L1077

**Comment by grondinjc** [comment]
From what I see, it is mentionned in the following RFC
https://datatracker.ietf.org/doc/html/draft-ietf-mmusic-sdp-bundle-negotiation-54#section-7.3

Sample offer : 
https://datatracker.ietf.org/doc/html/draft-ietf-mmusic-sdp-bundle-negotiation-54#section-7.2.2
![image](https://user-images.githubusercontent.com/5757055/141146790-d01f0d8e-b4fd-4ad7-afc1-375bf0bdb6db.png)

Sample answer to the sample offer : 
https://datatracker.ietf.org/doc/html/draft-ietf-mmusic-sdp-bundle-negotiation-54#section-7.3.4
![image](https://user-images.githubusercontent.com/5757055/141146962-49dea5b0-1892-4b1d-ba35-2fbba9707d81.png)

**Comment by grondinjc** [comment]
Relevant information that I forgot, this was found on SIPSorcery version="5.3.0-pre"

**Comment by grondinjc** [comment]
Will wait on the next nuget version to be publish to close the issue. Thanks

**Comment by grondinjc** [comment]
Issue is confirmed to be fixed with version 6.0.1-pre
